### DISCUSSION TITLE
File.creationDate() on Android

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#1363] [Q&A]
On Android `File.creationDate` returns same result as `File.modificationDate`. Android documenation states that from API 26 is possible to get `creationTime()` for files with [`BasicFileAttributes`](https://developer.android.com/reference/java/nio/file/attribute/BasicFileAttributes).

Is there any chance to include it in `File.creationDate` response?

**Comment by MalacTheLittle** [comment]
@ajwfrost Have any thoughts about this?

**Reply by ajwfrost** [reply]
Hi @MalacTheLittle  - we just looked at this. There's an interesting comment under the Java definition for `File.creationDate`:
> If the file system implementation does not support a time stamp to indicate the time when the file was created then this method returns an implementation specific default value, typically the last-modified-time or a FileTime representing the epoch (1970-01-01T00:00:00Z).

When we looked at the current code, it's not actually calling any Java APIs, it's just using the underlying Linux filesystem implementations and the `stat()` function:
https://linux.die.net/man/2/stat

So .. we could potentially look at calling out to Java to get the creation date but this is going to add some time to the call and we don't know whether this will actually give us a better value: it may be that the existing C++ implementation with `stat.st_ctime` may be returning the same thing here... 

Are you seeing obviously-wrong values coming out in these date fields? as, the code does treat `creationDate` separately from `modificationDate` so if the values are the same, it's because the `stat.st_ctime` is coming back with the same timestamp as `stat.st_mtime` (for modification time)...

thanks

**Reply by MalacTheLittle** [reply]
I was testing on Huawei p30 (API level 29) and Huawei p10 (API 28). All files and folders from `documentDirectory` returns same value for `modificationDate` and `creationDate`.
I'm not experienced in C++ or Java, just found that info and thought it has some potential :)

Here's screenshot of my test app result:
![image](https://user-images.githubusercontent.com/56675407/142924360-b3c7775d-a7e3-4616-8e66-59ad988c8ded.png)

**Reply by MalacTheLittle** [reply]
I've also tried to modify test file with FileMode.WRITE and FileMode.UPDATE and got same results. Only if I manually rename file then creationDate is untouched and modificationDate is updated.
